homeassistant/components/technove/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from technove import Station as TechnoVEStation
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfElectricCurrent,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import TechnoVEDataUpdateCoordinator
+from .models import TechnoVEEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TechnoVESensorEntityDescription(SensorEntityDescription):
+    """Describes TechnoVE sensor entity."""
+
+    value_fn: Callable[[TechnoVEStation], StateType]
+
+
+SENSORS: tuple[TechnoVESensorEntityDescription, ...] = (
+    TechnoVESensorEntityDescription(
+        key="voltage_in",
+        translation_key="voltage_in",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.voltage_in,
+    ),
+    TechnoVESensorEntityDescription(
+        key="voltage_out",
+        translation_key="voltage_out",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.voltage_out,
+    ),
+    TechnoVESensorEntityDescription(
+        key="max_current",
+        translation_key="max_current",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.max_current,
+    ),
+    TechnoVESensorEntityDescription(
+        key="max_station_current",
+        translation_key="max_station_current",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.max_station_current,
+    ),
+    TechnoVESensorEntityDescription(
+        key="current",
+        translation_key="current",
```

Comment:
Can use device class translations, so you can remove this

Reply:
Done.